### PR TITLE
Update to v22.1

### PR DESCRIPTION
**Pull request by SvetlanaMikheeva**

### CODE REVIEW

File: JS/README.md [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
+Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory.
 
 ## Running unit tests
 
 Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.io).
 
 ## Running end-to-end tests
 
-Run `ng e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).
+Run `ng e2e` to execute the end-to-end tests via a platform of your choice. To use this command, you need to first add a package that implements end-to-end testing capabilities.
```
Comment (dmitry-eliseev-devexpress) [inline JS/README.md]: No exceptions:

https://tw.devexpress.devx/language-guidelines/misused-words/via/

> capabilities

Sounds like a clericalism to me.

> execute **the** end-to-end tests

The definite article places extra emphasis on these end-to-end tests because we are not referring to an earlier introduction here. Therefore, no article is necessary.

```suggestion
Run `ng e2e` to execute end-to-end tests on a platform of your choice. To use this command, you need to first add a package that implements the end-to-end testing functionality.
```

File: JS/README.md [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
 Run `ng generate component component-name` to generate a new component. You can also use `ng generate directive|pipe|service|class|guard|interface|enum|module`.
 
 ## Build
 
-Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory. Use the `--prod` flag for a production build.
+Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory.
 
 ## Running unit tests
 
 Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.io).
```
Comment (dmitry-eliseev-devexpress) [inline JS/README.md]: Karma is a testing platform, is it not? As before with "via" -- no exceptions can be made here: https://tw.devexpress.devx/language-guidelines/misused-words/via/

```suggestion
Run `ng test` to execute the unit tests on [Karma](https://karma-runner.github.io).
```

File: JS/src/app/richedit/richedit.component.ts [hunk 3]
```diff
@@ -11,48 +8,48 @@ import { AfterViewInit, Component, ElementRef, OnDestroy } from '@angular/core';
...
 
-    homeTab.removeItem(HomeTabItemId.DecreaseFontSize);
-    homeTab.removeItem(HomeTabItemId.IncreaseFontSize);
+    homeTab!.removeItem(HomeTabItemId.DecreaseFontSize);
+    homeTab!.removeItem(HomeTabItemId.IncreaseFontSize);
 
-    homeTab.removeItem(HomeTabItemId.ChangeStyle);
+    homeTab!.removeItem(HomeTabItemId.ChangeStyle);
 
     // remove element from drop down list
```
Comment (dmitry-eliseev-devexpress) [inline JS/src/app/richedit/richedit.component.ts]: Generally, code comments follow the same grammar rules as the rest of your text. And yes, consistency matters.

```suggestion
    // remove an element from a drop-down list
```

File: JS/src/app/richedit/richedit.component.ts [hunk 4]
```diff
@@ -11,48 +8,48 @@ import { AfterViewInit, Component, ElementRef, OnDestroy } from '@angular/core';
...
-    // fileTab.removeItem(FileTabItemId.DownloadRtf);
-    // fileTab.removeItem(FileTabItemId.DownloadTxt);
+    // fileTab!.removeItem(FileTabItemId.DownloadRtf);
+    // fileTab!.removeItem(FileTabItemId.DownloadTxt);
 
-    // allow only download as docx. Create out item from default items
-    const downloadDocxItem = fileTab.getItem(FileTabItemId.DownloadDocx) as RibbonSubMenuItem;
-    const downloadItem = fileTab.getItem(FileTabItemId.Download) as RibbonMenuItem;
-    fileTab.removeItem(FileTabItemId.Download);
+    // allow download in docx format only. Create an item based on the default item options.
```
Comment (dmitry-eliseev-devexpress) [inline JS/src/app/richedit/richedit.component.ts]: Allow whom? Download what? A document, most likely. As I said before, comments follow the same grammar rules as the rest. Right now, the first "sentence" is just a disjoint set of words that contrasts with the second sentence you laboriously corrected.

Please rewrite the first part.

File: README.md [hunk 5]
```diff
@@ -3,40 +3,46 @@
 [![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T848130)
 [![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
 <!-- default badges end -->
-*Files to look at:*
-* [richedit.component.ts](./JS/src/app/richedit/richedit.component.ts)
+# Rich Text Editor for ASP.NET Core - How to customize the built-in ribbon in an Angular application
```
Comment (dmitry-eliseev-devexpress) [inline README.md]: Again, should we capitalize Ribbon? I have a feeling (and a memory) that we should. The same also goes for the rest of this topic. Also, Ribbon what? the built-in Ribbon UI, more precisely.

For some reason, I'm unable to edit this line directly.

File: README.md [hunk 6]
```diff
@@ -3,40 +3,46 @@
...
 
-# RichEdit for Angular - How to customize the built-in ribbon
+This example is a ready-to-use Angular client application that demonstrates how to customize the DevExpress [Rich Text Editor](https://docs.devexpress.com/AspNetCore/400373/office-inspired-controls/controls/rich-edit)'s ribbon.
```
Comment (dmitry-eliseev-devexpress) [inline README.md]:
```suggestion
This example is a ready-to-use Angular client application that demonstrates how to customize the DevExpress [Rich Text Editor](https://docs.devexpress.com/AspNetCore/400373/office-inspired-controls/controls/rich-edit)'s Ribbon UI.
```

File: README.md [hunk 7]
```diff
@@ -3,40 +3,46 @@
...
...lbar that displays command items on multiple tabs. Use the Rich Text Editor's [ribbon](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Options#js_devexpress_richedit_options_ribbon) option to access and customize the built-in ribbon. The [getTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_gettab_id_) and [getItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_getitem_id_) methods allow you to access the ribbon's tabs and their items. Call the [insertTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_inserttab_tab_) and [removeTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_removetab_tab_) methods to modify the collection of tabs. To modify the item collection, call the [insertItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_insertitem_item_) and [removeItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_removeitem_item_) methods.
```
Comment (dmitry-eliseev-devexpress) [inline README.md]: This paragraph has a series of issues:

1. Lacking capitalization of "Ribbon". I remember that we had a conversation where you agreed that we should capitalize it.
2. The first sentence does not outline the most important trait of the Ribbon UI - it groups commands according to their intended purpose.
3. Why do we use the mentioned `ribbon` option? To access and customize the built-in Ribbon UI. "Ribbon UI" is a more general term than "Ribbon" and includes all commands and UI elements.
4. the ribbon's tabs -- "Ribbon tabs" is way simpler and has the same effect.
5. The last sentence indirectly implies that we speak of items in a tab. Instead, we should speak of everything plainly and directly. As you may notice, I also outlined to what exactly the last mentioned methods belong.

**Bottom line** -- we should be as precise as possible in everything we do. If something can be understood incorrectly -- it surely will be. Here's my current take on this paragraph:

```suggestion
The Ribbon is a tabbed toolbar that groups command items on different tabs according to their purpose. Use the Rich Text Editor's [ribbon](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Options#js_devexpress_richedit_options_ribbon) option to access and customize the built-in Ribbon UI. The [getTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_gettab_id_) and [getItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_getitem_id_) methods allow you to access Ribbon tabs and their items. Call the [insertTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_inserttab_tab_) and [removeTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_removetab_tab_) methods to modify the collection of tabs. To modify the item collection in a tab, call its [insertItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_insertitem_item_) and [removeItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_removeitem_item_) methods.
```

File: README.md [hunk 8]
```diff
@@ -3,40 +3,46 @@
...
 
 ## Requirements
-* To use the RichEdit control in an Angular application, you need to have a [Universal, DXperience, or ASP.NET subscription](https://www.devexpress.com/buy/net/).
-* Versions of the devexpress npm packages should be identical (their major and minor versions should be the same).
+
+* To use the Rich Text Editor control in an Angular application, you need to have a [Universal, DXperience, or ASP.NET subscription](https://www.devexpress.com/buy/net/).
+* Versions of the DevExpress npm packages should match.
```
Comment (dmitry-eliseev-devexpress) [inline README.md]: I believe we should clarify this a bit. Whose versions should match? All DevExpress npm packages used in the current project, correct? If yes, we should be upfront with it.

File: README.md [hunk 9]
```diff
@@ -3,40 +3,46 @@
...
 
 ## Quick Start
 
-1. Open the **JS** folder. In the command prompt, download and install npm packages that are used in the application:
+Follow the steps below to run this application on your computer:
+
+1. Open the *JS* folder in the command prompt and run the following command to install npm packages used in the application:
```
Comment (dmitry-eliseev-devexpress) [inline README.md]: "Used in the application" implies definiteness for "npm packages". A definite article would have been too bold here, while "all" would have been a much better replacement:

```suggestion
1. Open the *JS* folder in the command prompt and run the following command to install all npm packages used in the application:
```

**Review by dmitry-eliseev-devexpress — COMMENTED** [review]
**Structure and Semantics**

This is a small topic on the Ribbon UI in Angular applications that serves its purpose. However, I feel a certain lack of precision in your wordings, especially when you explain the Ribbon UI concept itself.

The purpose of documentation is to answer questions -- both asked and unasked. That's why we so laboriously strive for precision, clarity, and unambiguity. If something in your text can be misunderstood -- it definitely will be. One day. Why give them this opportunity?

**Formatting**

I urge you to check once again if we should capitalize the word Ribbon. If memory serves, we already had such a conversation where we came to a conclusion favoring capitalization. For now, I capitalized "Ribbon" everywhere I could, except for the property name and the heading that I cannot edit for some reason.

**Language**

In general, your language is good and lives up to DevExpress standards. However, I'd like to see a bit more precision and unambiguity. Please also mind that code comments follow the same grammar rules as the rest of content we produce -- no comment should be a disjoint set of words. A comment can be only a sentence or an incomplete sentence (in certain cases). This aspect is even more important considering the fact that a half of one comment is written properly. Consistency and grammar matter. Everywhere.

I also recommend you to scan the content you receive from developers for "stop-words", such as "via", "represents", etc.

**Bottom Line**

In general, this is a good topic. However, we should to clarify a couple of things before moving forward.

File: README.md (hunk 5 above)
Comment (SvetlanaMikheeva) [reply README.md]:
```suggestion
# Rich Text Editor for ASP.NET Core - How to customize the built-in Ribbon UI in an Angular application
```

File: README.md (hunk 8 above)
Comment (SvetlanaMikheeva) [reply README.md]: You are right, versions of all DevExpress npm packages used in the same project should match.

```suggestion
* Versions of all DevExpress npm packages used in this application should match.
```

Comment (dmitry-eliseev-devexpress) [reply README.md]: OK.. but now the whole sentence looks a bit unnatural. What do you think of the following version?

```suggestion
* All Devexpress npm packages used in this application should have the same version.
```

File: JS/README.md (hunk 1 above)
Comment (SvetlanaMikheeva) [reply JS/README.md]: This Readme file was generated by Angular automatically when I created an Angular project. Maybe we could leave it as it is?

Comment (dmitry-eliseev-devexpress) [reply JS/README.md]: Does it mean that if we change this line, it will change back on the next update?

Comment (SvetlanaMikheeva) [reply JS/README.md]: No, it will not change back on the next update. It's just not our learning material, but we can improve it.

Comment (dmitry-eliseev-devexpress) [reply JS/README.md]: OK. I'll let you decide what to do with automatically generated text. Either way, I proposed my changes -- you can either apply or discard them. Or come up with something better instead.

File: JS/src/app/richedit/richedit.component.ts (hunk 4 above)
Comment (SvetlanaMikheeva) [reply JS/src/app/richedit/richedit.component.ts]:
```suggestion
    // create an item based on the default item options to allow users to download documents in docx format only
```

Comment (dmitry-eliseev-devexpress) [reply JS/src/app/richedit/richedit.component.ts]: Looks good to me. Thank you.

**Review by dmitry-eliseev-devexpress — APPROVED** [review]
Looks like we sorted everything out.

File: JS/README.md [hunk 10]
```diff
@@ -1,27 +1,27 @@
...
 
 ## Build
 
-Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory. Use the `--prod` flag for a production build.
+Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory.
 
 ## Running unit tests
 
-Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.io).
+Run `ng test` to execute the unit tests on [Karma](https://karma-runner.github.io).
```
Comment (arminal) [inline JS/README.md]:
```suggestion
Run `ng test` to execute unit tests on [Karma](https://karma-runner.github.io).
```

File: JS/README.md [hunk 11]
```diff
@@ -1,27 +1,27 @@
...
 
 ## Running end-to-end tests
 
-Run `ng e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).
+Run `ng e2e` to execute end-to-end tests on a platform of your choice. To use this command, you need to first add a package that implements the end-to-end testing functionality.
 
 ## Further help
 
-To get more help on the Angular CLI use `ng help` or go check out the [Angular CLI README](https://github.com/angular/angular-cli/blob/master/README.md).
+To get more help on the Angular CLI use `ng help` or go check out the [Angular CLI Overview and Command Reference](https://angular.io/cli) page.
```
Comment (arminal) [inline JS/README.md]:
```suggestion
To get more help on the Angular CLI, use `ng help` or check out the [Angular CLI Overview and Command Reference](https://angular.io/cli) page.
```

File: README.md [hunk 12]
```diff
@@ -3,40 +3,46 @@
...
 
 ## Requirements
-* To use the RichEdit control in an Angular application, you need to have a [Universal, DXperience, or ASP.NET subscription](https://www.devexpress.com/buy/net/).
-* Versions of the devexpress npm packages should be identical (their major and minor versions should be the same).
+
+* To use the Rich Text Editor control in an Angular application, you need to have a [Universal, DXperience, or ASP.NET subscription](https://www.devexpress.com/buy/net/).
```
Comment (arminal) [inline README.md]:
```suggestion
* To use the Rich Text Editor control in an Angular application, a [Universal, DXperience, or ASP.NET subscription](https://www.devexpress.com/buy/net/) is required.
```

File: README.md [hunk 13]
```diff
@@ -3,40 +3,46 @@
...
+* All Devexpress npm packages used in this application should have the same version.
```
Comment (arminal) [inline README.md]: the same version of what? specify.

Comment (SvetlanaMikheeva) [reply README.md]: Versions of all DevExpress npm packages should be the same. Be default, in this example the 22.1.9 version of the "devexpress-richedit" npm package is used. If you want to add another our package (for instance, the "devextreme" package) to this application, you must use the 22.1.9 version of this package.

Comment (arminal) [reply README.md]:
```suggestion
* All Devexpress npm packages used in this application should come from the same npm package version.
```
